src/engine/Unit_engine.ml

```diff
@@ -591,10 +591,19 @@ let tainting_test lang rules_file file =
                lazy_ast_and_errors = lazy (ast, []);
              }
            in
-           let res, _debug =
-             Match_tainting_mode.check_rule rule (fun _ _ -> ()) xconf xtarget
+           let results =
+             Match_tainting_mode.check_rules [ rule ]
+               (fun _ _ -> ())
+               xconf xtarget
+               (fun _rule f -> f ())
            in
-           res.matches)
+           match results with
+           | [ res ] -> res.matches
+           (* By construction, `check_rules` should only return the same number of results as rules it
+              was initially given.
+              So this case is impossible.
+           *)
+           | __else__ -> raise Impossible)
```

Comment:
I don't mind it here because it's test code, but otherwise I would have used `| [] | _::_::_ -> raise Impossible` instead.